utility-emissions-channel/docker-compose-setup/fabric-config/core.yaml

```diff
@@ -310,7 +310,7 @@ peer:
         # Settings for the PKCS#11 crypto provider (i.e. when DEFAULT: PKCS11)
         PKCS11:
             # Location of the PKCS11 module library
-            Library:
+            Library: /usr/local/lib/softhsm/libsofthsm2.so
```

Comment:
@brioux by this are you trying to use HSM for peer also ? If yes then you will have to change `default=PKCS11` and add these field also to `pkcs11:` example

```yaml
bccsp:
  default: PKCS11
  pkcs11:
    Library: /etc/hyperledger/fabric/libsofthsm2.so
    Pin: "71811222"
    Label: fabric
    hash: SHA2
    security: 256
    Immutable: false
```

Reply:
Thanks for the response.
Yes, I have looked at the pkcs11 config for the peer HSM but not fully implemented. Will continue working on it this week. 

This is a first step to get softHSM running on the docker API image, generate the tokens, and setup the HsmX509Provider for wallets used in the register/enroll user functions...